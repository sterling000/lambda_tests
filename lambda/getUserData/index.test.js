// stubs
const stubs = require('./stubs');

// mocks
const getValidMockUser = () => {
    return stubs.userData;
};

const getMissingMockUser = () => {
    return {};
};

jest.mock('node-fetch', () => {
    const stubs = require('./stubs')
    const generateResponse = () => {
        return {
            status: 200,
            json: () => ({results: stubs.userData.Item.decks})
        };
    };

    return jest
        .fn()
        .mockResolvedValue(generateResponse());
});

// target code for tests
const { handler } = require('./index.js');

// tests
describe('handler', () => {
    test('the handler function should work',async () => {
        jest.resetModules();
        jest.doMock('aws-sdk', () => {
            return {
                ...jest.requireActual("aws-sdk"),
                DynamoDB: {
                    DocumentClient: jest.fn(() => ({get: () => ({promise: getValidMockUser})}))
                }
            }
        });
        const response = await handler({pathParameters:{id:'Wildcard'}}, {});
        expect(response.body).toBeDefined();
        expect(response.statusCode).toBeDefined();
    });

    test('the response should be successful', async () => {
        jest.resetModules();
        jest.doMock('aws-sdk', () => {
            return {
                ...jest.requireActual("aws-sdk"),
                DynamoDB: {
                    DocumentClient: jest.fn(() => ({get: () => ({promise: getValidMockUser})}))
                }
            }
        });
        const response = await handler({pathParameters:{id:'Wildcard'}}, {});
        expect(response.statusCode).toEqual(200);
    });

    test('the response body should match the snapshot', async () => {
        jest.resetModules();
        jest.doMock('aws-sdk', () => {
            return {
                ...jest.requireActual("aws-sdk"),
                DynamoDB: {
                    DocumentClient: jest.fn(() => ({get: () => ({promise: getValidMockUser})}))
                }
            }
        });
        const response = await handler({pathParameters:{id:'Wildcard'}}, {});
        expect(response.statusCode).toEqual(200);
        const body = JSON.parse(response.body);
        expect(body.decks).toMatchSnapshot();
    });

    test('the db was missing the user', async () => {
        jest.resetModules();
        jest.doMock('aws-sdk', () => {
            return {
                ...jest.requireActual('aws-sdk'),
                DynamoDB: {
                    DocumentClient: jest.fn(() => ({
                        get: () => ({promise: getMissingMockUser}),
                        put: () => ({promise: jest.fn().mockImplementation(() => {return Promise.resolve({})})
                    }),  
                }))
                }
            }            
        });
        
        const response = await handler({pathParameters:{id:'Wildcard'}}, {});
        expect(response.statusCode).toEqual(202);
        const body = JSON.parse(response.body);
        expect(body.decks).toMatchSnapshot();
    });

    test('the db was missing the user and archidekt failed', async () => {
        jest.resetModules();
        jest.doMock('aws-sdk', () => {
            return {
                ...jest.requireActual('aws-sdk'),
                DynamoDB: {
                    DocumentClient: jest.fn(() => ({
                        get: () => ({promise: getMissingMockUser}),
                        put: () => ({promise: jest.fn()})
                    })),  
                }
            }
        });
        jest.doMock('node-fetch', ()=> {
            return {
                ...jest.requireActual('node-fetch'),
                fetch: () => ({promise: getMissingMockUser})
            }
        });
        const response = await handler({pathParameters:{id:'Wildcard'}}, {});
        expect(response.statusCode).toEqual(202);
        const body = JSON.parse(response.body);
        expect(body.decks).toMatchSnapshot();
    });
});