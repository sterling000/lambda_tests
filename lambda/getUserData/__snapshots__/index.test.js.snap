// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler the db was missing the user 1`] = `
Array [
  Object {
    "colors": Object {
      "G": 16,
      "R": 15,
      "U": 24,
    },
    "id": 567796,
    "name": "Xyris",
    "private": false,
    "updatedAt": "2020-06-21T20:34:51.995446Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 31,
      "U": 25,
    },
    "id": 567486,
    "name": "Zaxara",
    "private": false,
    "updatedAt": "2020-06-12T02:33:11.732940Z",
  },
  Object {
    "colors": Object {
      "G": 23,
      "R": 15,
      "U": 30,
    },
    "id": 547469,
    "name": "Kalamax Turns",
    "private": false,
    "updatedAt": "2020-06-13T03:03:32.261861Z",
  },
  Object {
    "colors": Object {
      "B": 19,
      "U": 40,
    },
    "id": 517923,
    "name": "Gyruda",
    "private": false,
    "updatedAt": "2020-06-12T02:31:34.083051Z",
  },
  Object {
    "colors": Object {
      "G": 36,
      "U": 17,
    },
    "id": 380487,
    "name": "Tatyova",
    "private": false,
    "updatedAt": "2020-06-12T02:04:51.213881Z",
  },
  Object {
    "colors": Object {
      "B": 29,
      "R": 21,
    },
    "id": 370431,
    "name": "Greven",
    "private": false,
    "updatedAt": "2020-06-13T01:00:56.451610Z",
  },
  Object {
    "colors": Object {
      "U": 40,
      "W": 21,
    },
    "id": 365563,
    "name": "Brago Blink",
    "private": false,
    "updatedAt": "2020-05-19T04:17:55.680890Z",
  },
  Object {
    "colors": Object {
      "B": 11,
      "G": 25,
      "U": 28,
    },
    "id": 216530,
    "name": "Kadena Morphs",
    "private": false,
    "updatedAt": "2020-04-11T02:42:06.771061Z",
  },
  Object {
    "colors": Object {
      "B": 34,
      "R": 28,
    },
    "id": 184378,
    "name": "Anje Wheels",
    "private": false,
    "updatedAt": "2020-05-19T20:18:10.671599Z",
  },
  Object {
    "colors": Object {
      "B": 59,
    },
    "id": 167531,
    "name": "K'rrik",
    "private": false,
    "updatedAt": "2020-04-23T03:35:37.099246Z",
  },
  Object {
    "colors": Object {
      "B": 15,
      "R": 18,
      "U": 32,
    },
    "id": 150583,
    "name": "Kess Consult",
    "private": false,
    "updatedAt": "2020-06-12T02:53:52.406596Z",
  },
  Object {
    "colors": Object {
      "W": 49,
    },
    "id": 115905,
    "name": "Brisela",
    "private": false,
    "updatedAt": "2020-04-12T01:19:02.284133Z",
  },
  Object {
    "colors": Object {
      "B": 7,
      "G": 30,
      "U": 18,
      "W": 17,
    },
    "id": 112514,
    "name": "Atraxa Counters",
    "private": false,
    "updatedAt": "2020-04-23T22:25:12.976916Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 20,
      "U": 32,
    },
    "id": 92696,
    "name": "Tasigur",
    "private": false,
    "updatedAt": "2020-06-12T02:05:17.396677Z",
  },
  Object {
    "colors": Object {
      "B": 20,
      "G": 22,
      "R": 4,
      "U": 27,
      "W": 6,
    },
    "id": 82731,
    "name": "Food Chain Niv",
    "private": false,
    "updatedAt": "2020-06-12T02:31:03.159763Z",
  },
  Object {
    "colors": Object {
      "G": 37,
      "R": 6,
      "U": 23,
    },
    "id": 82118,
    "name": "Harvest Animar cEDH",
    "private": false,
    "updatedAt": "2020-06-12T02:06:06.647873Z",
  },
]
`;

exports[`handler the db was missing the user and archidekt failed 1`] = `
Array [
  Object {
    "colors": Object {
      "G": 16,
      "R": 15,
      "U": 24,
    },
    "id": 567796,
    "name": "Xyris",
    "private": false,
    "updatedAt": "2020-06-21T20:34:51.995446Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 31,
      "U": 25,
    },
    "id": 567486,
    "name": "Zaxara",
    "private": false,
    "updatedAt": "2020-06-12T02:33:11.732940Z",
  },
  Object {
    "colors": Object {
      "G": 23,
      "R": 15,
      "U": 30,
    },
    "id": 547469,
    "name": "Kalamax Turns",
    "private": false,
    "updatedAt": "2020-06-13T03:03:32.261861Z",
  },
  Object {
    "colors": Object {
      "B": 19,
      "U": 40,
    },
    "id": 517923,
    "name": "Gyruda",
    "private": false,
    "updatedAt": "2020-06-12T02:31:34.083051Z",
  },
  Object {
    "colors": Object {
      "G": 36,
      "U": 17,
    },
    "id": 380487,
    "name": "Tatyova",
    "private": false,
    "updatedAt": "2020-06-12T02:04:51.213881Z",
  },
  Object {
    "colors": Object {
      "B": 29,
      "R": 21,
    },
    "id": 370431,
    "name": "Greven",
    "private": false,
    "updatedAt": "2020-06-13T01:00:56.451610Z",
  },
  Object {
    "colors": Object {
      "U": 40,
      "W": 21,
    },
    "id": 365563,
    "name": "Brago Blink",
    "private": false,
    "updatedAt": "2020-05-19T04:17:55.680890Z",
  },
  Object {
    "colors": Object {
      "B": 11,
      "G": 25,
      "U": 28,
    },
    "id": 216530,
    "name": "Kadena Morphs",
    "private": false,
    "updatedAt": "2020-04-11T02:42:06.771061Z",
  },
  Object {
    "colors": Object {
      "B": 34,
      "R": 28,
    },
    "id": 184378,
    "name": "Anje Wheels",
    "private": false,
    "updatedAt": "2020-05-19T20:18:10.671599Z",
  },
  Object {
    "colors": Object {
      "B": 59,
    },
    "id": 167531,
    "name": "K'rrik",
    "private": false,
    "updatedAt": "2020-04-23T03:35:37.099246Z",
  },
  Object {
    "colors": Object {
      "B": 15,
      "R": 18,
      "U": 32,
    },
    "id": 150583,
    "name": "Kess Consult",
    "private": false,
    "updatedAt": "2020-06-12T02:53:52.406596Z",
  },
  Object {
    "colors": Object {
      "W": 49,
    },
    "id": 115905,
    "name": "Brisela",
    "private": false,
    "updatedAt": "2020-04-12T01:19:02.284133Z",
  },
  Object {
    "colors": Object {
      "B": 7,
      "G": 30,
      "U": 18,
      "W": 17,
    },
    "id": 112514,
    "name": "Atraxa Counters",
    "private": false,
    "updatedAt": "2020-04-23T22:25:12.976916Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 20,
      "U": 32,
    },
    "id": 92696,
    "name": "Tasigur",
    "private": false,
    "updatedAt": "2020-06-12T02:05:17.396677Z",
  },
  Object {
    "colors": Object {
      "B": 20,
      "G": 22,
      "R": 4,
      "U": 27,
      "W": 6,
    },
    "id": 82731,
    "name": "Food Chain Niv",
    "private": false,
    "updatedAt": "2020-06-12T02:31:03.159763Z",
  },
  Object {
    "colors": Object {
      "G": 37,
      "R": 6,
      "U": 23,
    },
    "id": 82118,
    "name": "Harvest Animar cEDH",
    "private": false,
    "updatedAt": "2020-06-12T02:06:06.647873Z",
  },
]
`;

exports[`handler the response body should match the snapshot 1`] = `
Array [
  Object {
    "colors": Object {
      "G": 16,
      "R": 15,
      "U": 24,
    },
    "id": 567796,
    "name": "Xyris",
    "private": false,
    "updatedAt": "2020-06-21T20:34:51.995446Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 31,
      "U": 25,
    },
    "id": 567486,
    "name": "Zaxara",
    "private": false,
    "updatedAt": "2020-06-12T02:33:11.732940Z",
  },
  Object {
    "colors": Object {
      "G": 23,
      "R": 15,
      "U": 30,
    },
    "id": 547469,
    "name": "Kalamax Turns",
    "private": false,
    "updatedAt": "2020-06-13T03:03:32.261861Z",
  },
  Object {
    "colors": Object {
      "B": 19,
      "U": 40,
    },
    "id": 517923,
    "name": "Gyruda",
    "private": false,
    "updatedAt": "2020-06-12T02:31:34.083051Z",
  },
  Object {
    "colors": Object {
      "G": 36,
      "U": 17,
    },
    "id": 380487,
    "name": "Tatyova",
    "private": false,
    "updatedAt": "2020-06-12T02:04:51.213881Z",
  },
  Object {
    "colors": Object {
      "B": 29,
      "R": 21,
    },
    "id": 370431,
    "name": "Greven",
    "private": false,
    "updatedAt": "2020-06-13T01:00:56.451610Z",
  },
  Object {
    "colors": Object {
      "U": 40,
      "W": 21,
    },
    "id": 365563,
    "name": "Brago Blink",
    "private": false,
    "updatedAt": "2020-05-19T04:17:55.680890Z",
  },
  Object {
    "colors": Object {
      "B": 11,
      "G": 25,
      "U": 28,
    },
    "id": 216530,
    "name": "Kadena Morphs",
    "private": false,
    "updatedAt": "2020-04-11T02:42:06.771061Z",
  },
  Object {
    "colors": Object {
      "B": 34,
      "R": 28,
    },
    "id": 184378,
    "name": "Anje Wheels",
    "private": false,
    "updatedAt": "2020-05-19T20:18:10.671599Z",
  },
  Object {
    "colors": Object {
      "B": 59,
    },
    "id": 167531,
    "name": "K'rrik",
    "private": false,
    "updatedAt": "2020-04-23T03:35:37.099246Z",
  },
  Object {
    "colors": Object {
      "B": 15,
      "R": 18,
      "U": 32,
    },
    "id": 150583,
    "name": "Kess Consult",
    "private": false,
    "updatedAt": "2020-06-12T02:53:52.406596Z",
  },
  Object {
    "colors": Object {
      "W": 49,
    },
    "id": 115905,
    "name": "Brisela",
    "private": false,
    "updatedAt": "2020-04-12T01:19:02.284133Z",
  },
  Object {
    "colors": Object {
      "B": 7,
      "G": 30,
      "U": 18,
      "W": 17,
    },
    "id": 112514,
    "name": "Atraxa Counters",
    "private": false,
    "updatedAt": "2020-04-23T22:25:12.976916Z",
  },
  Object {
    "colors": Object {
      "B": 16,
      "G": 20,
      "U": 32,
    },
    "id": 92696,
    "name": "Tasigur",
    "private": false,
    "updatedAt": "2020-06-12T02:05:17.396677Z",
  },
  Object {
    "colors": Object {
      "B": 20,
      "G": 22,
      "R": 4,
      "U": 27,
      "W": 6,
    },
    "id": 82731,
    "name": "Food Chain Niv",
    "private": false,
    "updatedAt": "2020-06-12T02:31:03.159763Z",
  },
  Object {
    "colors": Object {
      "G": 37,
      "R": 6,
      "U": 23,
    },
    "id": 82118,
    "name": "Harvest Animar cEDH",
    "private": false,
    "updatedAt": "2020-06-12T02:06:06.647873Z",
  },
]
`;
